framework/include/userobject/DomainUserObject.h

```diff
@@ -0,0 +1,235 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE includes
+#include "UserObject.h"
+#include "BlockRestrictable.h"
+#include "ThreeMaterialPropertyInterface.h"
+#include "NeighborCoupleable.h"
+#include "MooseVariableDependencyInterface.h"
+#include "TransientInterface.h"
+#include "RandomInterface.h"
+#include "ElementIDInterface.h"
+#include "MooseError.h"
+
+class MooseVariableFieldBase;
+
+namespace libMesh
+{
+class Elem;
+class QBase;
+}
+
```

Comment:
docstring

Reply:
@GiudGiud I will probably need more changes on this user object. So I will try addressing your comments here later when I push the changes up.